public/js/src/module/controller-webform.js

```diff
@@ -513,14 +514,23 @@ function _saveRecord( survey, draft = true, recordName, confirmed, errorMsg ) {
         } );
 }
 
+/**
+ * @type {Promise<void> | null}
+ */
+let autoSavePromise = null;
+
 function _autoSaveRecord() {
     // Do not auto-save a record if the record was loaded from storage
     // or if the form has enabled encryption
     if ( form.recordName || form.encryptionKey ) {
-        return Promise.resolve();
+        autoSavePromise = Promise.resolve();
+
+        return autoSavePromise.then( () => {
+            autoSavePromise = null;
+        } );
```

Comment:
I'm sure your understanding of promises is much more nuanced than mine, so please pardon the possibly obtuse question :blush: Why put this `null` assignment inside the callback? I consulted the [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) to try to be less ignorant, but I still didn't feel like I totally understood:
> An action can be assigned to an already "settled" promise. In that case, the action (if appropriate) will be performed at the first asynchronous opportunity. Note that promises are guaranteed to be asynchronous. Therefore, an action for an already "settled" promise will occur only after the stack has cleared and a clock-tick has passed. The effect is much like that of `setTimeout(action,10)`.

Reply:
To be entirely honest, after a few rounds through this issue/fix, I'm not sure I remember why I made this nullable. I suspect it's left over from a different approach 🙃 

It occurs to me now, though, that this branch should probably also wait for any previously assigned `autoSavePromise`.